CONTRIBUTING.md

```diff
@@ -49,15 +49,106 @@ If your pull request contains any of the below, it's less likely to be merged:
 **During pull request review**
 A core contributor will review your pull request and provide feedback. To ensure that there is not a large backlog of inactive PRs, the pull request will be marked as stale after two weeks of no activity. After another four days, it will be closed.
 
-### Architecture
-The architecture of project is enforced with validation diagrams and will be run automatically at build. See the `eng/DependencyValidation` project to update or adjust this diagram. The following is currently enforced:
+## Architecture
+
+### Startup and hosting
+
+Upgrade Assistant is a .NET 5 console application that uses `System.CommandLine` APIs to [parse command line arguments](src/cli/Microsoft.DotNet.UpgradeAssistant.Cli/Program.cs) and then executes commands. Logging is done using `Microsoft.Extensions.Logging` APIs with a Serilog logging provider writing logs to the console.
+
+Upgrade Assistant's most common command, currently, is the [`ConsoleUpgradeCommand`](src/cli/Microsoft.DotNet.UpgradeAssistant.Cli/Commands/Upgrade/ConsoleUpgradeCommand.cs). This command walks through a series of 'upgrade steps' to help move a user's project or solution from .NET Framework to .NET Core/5/6. `ConsoleUpgradeCommand` use a `Microsoft.Extensions.Hosting` [generic host](https://docs.microsoft.com/dotnet/core/extensions/generic-host) to register and execute upgrade services. The host uses the [`ConsoleRunner`](src/cli/Microsoft.DotNet.UpgradeAssistant.Cli/ConsoleRunner.cs) hosted service to execute the upgrade command when the host starts. All services are resolved from the [dependency injection container](https://docs.microsoft.com/dotnet/core/extensions/dependency-injection) setup by the command, which makes it easy for extensions to register services in addition to those registered by default. Key services for the upgrade command are described in the following sections.
+
+### Core upgrade components
+
+- [`UpgraderManager`](src/components/Microsoft.DotNet.UpgradeAssistant/UpgraderManager.cs) is the central Upgrade Assistant component responsible for managing the list of upgrade steps and progressing through them one at a time.
```

Comment:
This detail would make more sense within doc comments for the UpgraderManager so they can stay in sync.

This list here could be a good jumping place for it.